src/cascadia/TerminalApp/VsSetupConfiguration.h

```diff
@@ -0,0 +1,142 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- VsSetupConfiguration
+
+Abstract:
+- Encapsulates the Visual Studio Setup Configuration COM APIs
+
+Author(s):
+- Charles Willis - October 2020
+
+--*/
+
+#pragma once
+
+#include <wil/resource.h>
+#include "Setup.Configuration.h"
+
+namespace TerminalApp
+{
+    /// <summary>
+    /// See https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.setup.configuration?view=visualstudiosdk-2019
+    /// </summary>
+    class VsSetupConfiguration
+    {
+    private:
+        typedef wil::com_ptr<ISetupConfiguration2> ComPtrSetupQuery;
+        typedef wil::com_ptr<ISetupHelper> ComPtrSetupHelper;
+        typedef wil::com_ptr<ISetupInstance> ComPtrSetupInstance;
+        typedef wil::com_ptr<ISetupPropertyStore> ComPtrPropertyStore;
+        typedef wil::com_ptr<ISetupPackageReference> ComPtrPackageReference;
+        typedef wil::com_ptr<ISetupInstanceCatalog> ComPtrInstanceCatalog;
+
+    public:
+        struct VsSetupInstance
+        {
+            inline const static std::wstring_view kVersion16_3{ L"16.3.0.0" };

```

Comment:
Out of curiosity, why the `k` prefix?

Reply:
It's like mp, stands for konstant. I'll change those too